internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -427,14 +431,21 @@ func FindNewMachineSet(deployment *clusterv1.MachineDeployment, msList []*cluste
 	return nil
 }
 
+func shouldRolloutAfter(reconciliationTime *metav1.Time, rolloutAfter *metav1.Time, ms *clusterv1.MachineSet) bool {
```

Comment:
I know that Before handles nil gracefully but could we explicitly handle the rolloutAfter == nil case here? (same in the KCP code)